core/src/main/java/org/apache/iceberg/avro/PruneColumns.java

```diff
@@ -90,12 +92,18 @@ public Schema record(Schema record, List<String> names, List<Schema> fields) {
       } else if (fieldSchema != null) {
         hasChange = true;
         filteredFields.add(copyField(field, fieldSchema, fieldId));
+      } else if (emptyStructIds.contains(fieldId)) {
+        // This field does not require any known sub-fields but is required in the projection so keep it without
+        // any of it's subfields
+        hasChange = true;
+        Schema empty = AvroSchemaUtil.removeFields(field);
```

Comment:
Why use `removeFields` when this class already has `copyRecord`? Couldn't this just call `copyRecord(field.schema(), ImmutableList.of())`?

Reply:
CopyRecord does not work on UnionTypes :(